paymentsheet/src/main/java/com/stripe/android/link/confirmation/CompleteLinkFlow.kt

```diff
@@ -0,0 +1,150 @@
+package com.stripe.android.link.confirmation
+
+import com.stripe.android.core.strings.ResolvableString
+import com.stripe.android.link.LinkAccountUpdate
+import com.stripe.android.link.LinkAccountUpdate.Value.UpdateReason.PaymentConfirmed
+import com.stripe.android.link.LinkActivityResult
+import com.stripe.android.link.LinkDismissalCoordinator
+import com.stripe.android.link.LinkLaunchMode
+import com.stripe.android.link.LinkPaymentDetails
+import com.stripe.android.link.LinkPaymentMethod
+import com.stripe.android.link.account.LinkAccountManager
+import com.stripe.android.link.account.linkAccountUpdate
+import com.stripe.android.link.account.loadDefaultShippingAddress
+import com.stripe.android.link.confirmation.CompleteLinkFlow.Result
+import com.stripe.android.link.model.LinkAccount
+import com.stripe.android.link.withDismissalDisabled
+import com.stripe.android.model.ConsumerPaymentDetails
+import javax.inject.Inject
+import com.stripe.android.link.confirmation.Result as LinkConfirmationResult
+
+/**
+ * Completes the Link flow with a given payment based on the [LinkLaunchMode].
+ *
+ * - On [LinkLaunchMode.Full] and [LinkLaunchMode.Confirmation], confirms the payment.
+ * - On [LinkLaunchMode.PaymentMethodSelection], it returns the selected payment details.
+ */
+internal interface CompleteLinkFlow {
```

Comment:
This logic was duplicated across Wallet, CreateCard and **now** UpdateCard screens. Creates a common use case to avoid duplication.